### PR TITLE
AGENT-234: Take node0 selection from agent-config if it exists and se…

### PR DESCRIPTION
**Pull request by pawanpinjarkar**
If node0 IP is specified in agentConfig, it takes precedence over the selection from NMStateConfigs, otherwise, we keep the same heuristic as we have now to choose.

### CODE REVIEW

**Review by zaneb — COMMENTED** [review]
/approve

File: pkg/asset/agent/image/ignition.go [hunk 1]
```diff
@@ -112,9 +112,18 @@ func (a *Ignition) Generate(dependencies asset.Parents) error {
...
+	if agentConfigAsset.Config != nil && agentConfigAsset.Config.Spec.RendezvousIP != "" {
+		nodeZeroIP = agentConfigAsset.Config.Spec.RendezvousIP
+		logrus.Debug("RendezvousIP from the AgentConfig ", nodeZeroIP)
+
+	} else {
+		nodeZeroIP, err := manifests.GetNodeZeroIP(agentManifests.NMStateConfigs)
+		logrus.Debug("RendezvousIP from the NMStateConfig ", nodeZeroIP)
+		if err != nil {
+			return err
+		}
```
Comment (andfasano) [inline pkg/asset/agent/image/ignition.go:126]: It's be nice to have a test for this case

**Pull request merged**